code/API_definitions/location-subscriptions.yaml

```diff
@@ -0,0 +1,640 @@
+        eventType: 
+          $ref: "#/components/schemas/EventType"
+      required:
+        - device
+        - area
+        - eventType
+    
+    EventType:
+      description: |
+        AREA_ENTERED - Location event triggered when the device entered the given area
+        AREA_LEFT - Location event triggered when the device left the given area
+        SUBSCRIPTION_ENDS - It is inform listener request that subscription terminates.
+      type: string
+      enum:
+        - AREA_ENTERED
+        - AREA_LEFT
+        - SUBSCRIPTION_ENDS
+    
+    NotificationUrl:
+      type: string
+      example: "https://application-server.com"
+      description: https callback address where the event notification must be POST-ed
+    
+    NotificationAuthToken:
+      type: string
+      example: "c8974e592c2fa383d4a3960714"
+      description: Authentication token for callback API 
+    
+    CreateEventSubscription:
+      description: The request for creating a Location Event Subscription
+      type: object
+      properties:
+        webhook:
+          $ref: '#/components/schemas/Webhook'
+        subscriptionDetail:
+          $ref: '#/components/schemas/SubscriptionDetail'
+        subscriptionExpireTime:
+          type: string
+          format: date-time
+          example: 2023-01-17T13:18:23.682Z
+          description: The time when the location-tracking has to be terminated (e.g. in 14 days from now on). Provided in date-time format.
+      required:
+        - webhook
+        
+    SubscriptionInfo:
+      description: Represents a area-location event subscription.
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/CreateEventSubscription'
+        - type: object
+          properties:
+            eventSubscriptionId:
+              $ref: '#/components/schemas/EventSubscriptionId'
+            startsAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking started (e.g. in 7 days from now on). Provided in date-time format.
+            expiresAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking expired (e.g. in 7 days from now on). Provided in date-time format.
+          required:
+            - eventSubscriptionId
+            - webhook
+  
+    EventTime:
+      format: date-time
+      type: string
+      description: The time when the event was notified.
+    
+    EventNotifications:
+      description: The event notification callback.
+      type: object
+      required:
+        - event
+      properties:
+        eventSubscriptionId:
+          $ref: '#/components/schemas/EventSubscriptionId'
+        event:
+          $ref: '#/components/schemas/Event'
+    
+    Event:
+      description: The event being notified
+      anyOf:
```

Comment:
There was a discussion in QoD to favour allOf vs anyOf to model the event, due to better compatibility with some generators and coding languages, you can check the PR here: https://github.com/camaraproject/QualityOnDemand/pull/177

This was proposed by Orange, @bigludo7